packages/lang-server/src/workspace.ts

```diff
@@ -203,20 +208,22 @@ export class EditorWorkspace {
   private async runAggregatedSetOperation(): Promise<void> {
     if (this.hasPendingUpdates() && this.workspace !== undefined) {
       const opDeletes = this.pendingDeletes
-      const opNaclFiles = this.pendingSets
+      const opUpdates = this.pendingSets
       this.pendingDeletes = new Set<string>()
       this.pendingSets = {}
       // We start by running all deleted
       const removeChanges = (!_.isEmpty(opDeletes))
         ? await this.workspace.removeNaclFiles(...opDeletes)
         : []
       // Now add the waiting changes
-      const updateChanges = (!_.isEmpty(opNaclFiles))
-        ? await this.workspace.setNaclFiles(...Object.values(opNaclFiles))
+      const updateChanges = (!_.isEmpty(opUpdates))
+        ? await this.workspace.setNaclFiles(...Object.values(opUpdates))
         : []
       if (this.wsErrors !== undefined) {
         const validation = await this.getValidationErrors(
-          [...opDeletes, ...Object.keys(opNaclFiles)], [...removeChanges, ...updateChanges]
+          [...opDeletes, ...Object.keys(opUpdates)]
+            .filter(f => this.workspace.envOfFile(f)),
```

Comment:
should this have been 
```suggestion
            .filter(this.workspace.isWorkspaceFile),
```
?